### ISSUE TITLE
[Rule idea] No empty non void elements

### ISSUE THREAD

**Issue by ansmonjol** [ember-template-lint/ember-template-lint#726]
Hello,

I would love to implement a new rule to avoid non-empty void elements in templates.

Something that could lead to raise an error on `<p></p>` or `<div></div>` for example. 
Of course nothing would happen on `<p>{{myVar}}</p>` or any void elements (`<br >`).

Do you think it's ok to call this rule `no-empty-non-void-elements` ?

In order to submit an acceptable final MR, do you have some auto generated files (like Changelog ?) or stuff I must perform during new rule implementation ? In plus than Documentation and tests.

Thanks

**Comment by rwjblue** [comment]
I don't fully grok the request here, can you update the description to show more good and bad examples?

**Comment by ansmonjol** [comment]
Done ! let me know if that's cleat enough :)

**Comment by rwjblue** [comment]
Gotcha, I was misinterpreting the term "void" in your original post. I typically use the term "void elements" to refer to things like:

```html
<input>
<br>
<img>
```

Which apparently are actually called [empty elements](https://developer.mozilla.org/en-US/docs/Glossary/empty_element). 

I'd vaguely prefer to avoid confusing my future self by not using `void` in the name here, what do you think about something like `blacklist-childless-elements`?

**Comment by ansmonjol** [comment]
Great ! Thanks for your comment.

I'm a bit afraid by the `blacklist` wording here WDYT ?
I was trying to keep consistency with the `no-` rule name. Also, the 'config' item list is directly inspired from the `lint-self-closing-void-elements` rule.

I propose to rename the config list to `BLOCK_TAGS` and the linter rule to `no-childless-elements` if that sounds good to you ?

**Comment by rwjblue** [comment]
Seems great!

**Comment by ansmonjol** [comment]
Hello @rwjblue, you can find the related PR https://github.com/ember-template-lint/ember-template-lint/pull/735

Please let me know if I forgot anything or something is not clear enough !

**Comment by MelSumner** [comment]
'<div></div>` is pretty common to leave blank for reasons. I'd probably recommend removing that from the disallowed list.
